src/screens/HomeScreen.tsx

```diff
@@ -109,14 +113,15 @@ const HomeScreen = ({navigation}: any) => {
     set_is_loading(false);
   }, []);
 
-  console.log(upcoming_movies_list);
+  const handle_selected_movie = (id: number) => {
+    navigation.push('MovieDetail', {
+      movie_id: id,
+    });
+  };
 
   if (is_loading) {
     return (
-      <ScrollView
-        style={styles.container}
-        bounces={false}
-        contentContainerStyle={styles.scroll_view_container}>
+      <ScrollView style={styles.container} bounces={false}>
         <StatusBar hidden />
         <View style={styles.search_bar_input}>
           <SearchBar search_functionality={search_functionality} />
```

Comment:
`<SearchBar>` component is rendering twice.
If the component has to be visible regardless of the condition, then prefer to impose the condition only on the set of components that need to be rendered dynamically.